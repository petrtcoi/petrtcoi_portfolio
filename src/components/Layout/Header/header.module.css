.headerblock {
  display: flex;
  padding: 40px 40px 80px 40px;
  justify-content: space-between;
  align-items: center;
}
@media screen and (max-width: 800px) {
  .headerblock {
    padding: 30px 20px 50px 20px;
  }
}

.title {
  font-size: 1.4rem;
  line-height: 1.4rem;
  font-weight: 600;
}
.subtitle {
  font-size: 0.8rem;
  line-height: 1rem;
  font-weight: 200;
}
.logo {
  fill: var(--color-accent);
}
.burgerbutton {
  border: 0;
  background-color: var(--color-background);
  transition: background-color var(--basic-duration);
}
