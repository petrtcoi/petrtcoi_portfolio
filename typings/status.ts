export enum MenuStatus {
    open = 'open',
    closed = 'closed'
}