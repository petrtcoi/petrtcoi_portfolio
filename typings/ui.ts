export enum MenuStatus {
    open = 'open',
    closed = 'closed'
}

export enum Theme {
    light = 'light',
    dark = 'dark'
}