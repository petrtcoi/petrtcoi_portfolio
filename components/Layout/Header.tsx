import { FC, useState } from "react"
import BurgerMenuIcon from "../../assets/icons/BurgerMenuIcon"
import { MenuStatus } from "../../typings/status"

import ReactLogo from "../ReactLogo/ReactLogo"

import styles from './css/layout.module.css'
import PopupMenu from "./PopupMenu"


type HeaderProps = {}

const Header: FC<HeaderProps> = (_props) => {

    const [menuStatus, setMenuStatus] = useState<MenuStatus>(MenuStatus.closed)

    return (
        <header data-testid="header">


            <div
                data-testid="react-logo"
            >
                <ReactLogo size={ 50 } />
            </div>


            <div
                data-testid="header-title"
                className={ styles["header-title"] } >
                Petr Tcoi
            </div>


            <div
                data-testid="popupmenu-burger-button"
                onClick={ () => setMenuStatus(MenuStatus.open) }
            >
                <BurgerMenuIcon size={ 25 } className={ "icon-button" } />
            </div>


            <div data-testid="popup-menu"
                data-popupmenu-status={ menuStatus }
                className={ styles['popup-menu'] }
                style={ { opacity: menuStatus === MenuStatus.closed ? 0 : 100 } }
            >
                <PopupMenu
                    closeMenuFunc={ () => setMenuStatus(MenuStatus.closed) }
                />
            </div>


        </header>
    )
}

export default Header


