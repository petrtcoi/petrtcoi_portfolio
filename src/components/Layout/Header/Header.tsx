import React from "react"
import Link from "next/link"

import BurgerMenuIcon from "../../../assets/icons/BurgerMenuIcon"
import PopupMenu from "../PopupMenu/PopupMenu"
import ReactLogo from "../../../assets/icons/ReactLogo"


import styles from './header.module.css'
import { MenuStatus } from "../../../assets/types/ui.type"



type NavbarProps = {}

const Navbar: React.FC<NavbarProps> = (_props) => {

  const [menuStatus, setMenuStatus] = React.useState<MenuStatus>(MenuStatus.closed)


  return (
    <header data-testid={ "layout-header" } className={ styles.headerblock }>

      {/* LOGO */ }
      <Link href="/">
        <ReactLogo size={ 50 } className={ styles.logo } />
      </Link>


      {/* TITLE */ }
      <h1 style={{textAlign: "center"}}>
        <div className={ styles.title }>
        Petr Tcoi
        </div>
        <div className={ styles.subtitle }>
          портфолио
        </div>
        
      </h1>


      {/* BURGER MENU */ }
      <button
        aria-label={ "Открыть меню" }
        data-state-hide={ menuStatus === MenuStatus.open ? "true" : "false" }
        className={ styles.burgerbutton }
        onClick={ () => setMenuStatus(MenuStatus.open) }
      >
        <BurgerMenuIcon size={ 25 } className={ "svg__button" } />
      </button>


      {/* POPUPMENU */ }
      <PopupMenu
        menuStatus={ menuStatus }
        closeMenuFunc={ () => setMenuStatus(MenuStatus.closed) }
      />

    </header>
  )

}

export default Navbar




