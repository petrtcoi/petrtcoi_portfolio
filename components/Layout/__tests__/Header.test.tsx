import { render, screen } from '@testing-library/react'

import userEvent from '@testing-library/user-event'
import Header from '../Header'
import { MenuStatus } from '../../../typings/ui'





describe('Header', () => {


    describe('Popup Menu', () => {


        test('Выводится логотип React', () => {
            render(<Header />)
            expect(screen.getByTestId('react-logo')).toBeVisible()
        })

        test('Выводится Заголовок в шапке', () => {
            render(<Header />)
            expect(screen.getByTestId('header-title')).toBeVisible()
        })


        test('Скриншот тест для верстки', () => {
            const container = render(<Header />)
            expect(container).toMatchSnapshot()
        })


        test('Есть бургер, кнопка для открытия меню', () => {
            render(<Header />)
            expect(screen.getByTestId('popupmenu-burger-button')).toBeVisible()
        })

        test('Popup меню скрыто и имеет статус закрыто', () => {
            render(<Header />)
            const popupMenu = screen.getByTestId('popup-menu')
            expect(popupMenu.dataset.popupmenuStatus).toEqual(MenuStatus.closed)
            expect(popupMenu).not.toBeVisible()
        })

        test('Клик по бургеру меняет статус меню на открыто и видно', async () => {
            render(<Header />)
            const popupMenu = screen.getByTestId('popup-menu')
            await userEvent.click(screen.getByTestId('popupmenu-burger-button'))
            expect(popupMenu.dataset.popupmenuStatus).toEqual(MenuStatus.open)
            expect(popupMenu).toBeVisible()
        })

        test('Клик по бургеру делает ставит на него свойстро "скрыть"', async () => {
            render(<Header />)
            const burgerButton = screen.getByTestId('popupmenu-burger-button')
            await userEvent.click(burgerButton)
            expect(burgerButton.dataset.shouldHide).toBe("true")
        })


        test('Клик по крестику меняет статус на закрыто и не видно', async () => {
            render(<Header />)
            const popupMenu = screen.getByTestId('popup-menu')

            await userEvent.click(screen.getByTestId('popupmenu-burger-button'))
            await userEvent.click(screen.getByTestId('popupmenu-close-button'))
            expect(popupMenu.dataset.popupmenuStatus).toEqual(MenuStatus.closed)
            expect(popupMenu).not.toBeVisible()
        })



    })
})