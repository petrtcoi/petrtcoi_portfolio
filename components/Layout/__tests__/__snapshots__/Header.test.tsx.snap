// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Popup Menu Скриншот тест для верстки 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        data-testid="header"
      >
        <div
          data-testid="react-logo"
        >
          <img
            alt="Логотип React. На котором пишет Petr Tcoi"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/images/icons/logo-react.svg"
            srcset="/images/icons/logo-react.svg 1x, /images/icons/logo-react.svg 2x"
            style="color: transparent; filter: invert(51%) sepia(85%) saturate(2665%) hue-rotate(322deg) brightness(111%) contrast(99%);"
            title="Логотип React. На котором хорошо пишет Petr Tcoi"
            width="50"
          />
        </div>
        <div
          class="header-title"
          data-testid="header-title"
        >
          Petr Tcoi
        </div>
        <div
          data-testid="popupmenu-burger-button"
        >
          <div
            aria-label="Кнопка открыть меню"
            class="icon-button"
          >
            <svg
              height="25"
              viewBox="0 0 100 75"
              width="25"
            >
              <rect
                height="15"
                width="100"
              />
              <rect
                height="15"
                width="100"
                y="30"
              />
              <rect
                height="15"
                width="100"
                y="60"
              />
            </svg>
          </div>
        </div>
        <div
          class="popup-menu"
          data-popupmenu-status="closed"
          data-testid="popup-menu"
          style="opacity: 0;"
        >
          <div>
            <div
              data-testid="popupmenu-close-button"
              style="display: inline-block;"
            >
              <div
                class="icon-button"
              >
                <svg
                  height="35"
                  viewBox="0 0 50 50"
                  width="35"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M 25 2 C 12.309534 2 2 12.309534 2 25 C 2 37.690466 12.309534 48 25 48 C 37.690466 48 48 37.690466 48 25 C 48 12.309534 37.690466 2 25 2 z M 25 4 C 36.609534 4 46 13.390466 46 25 C 46 36.609534 36.609534 46 25 46 C 13.390466 46 4 36.609534 4 25 C 4 13.390466 13.390466 4 25 4 z M 32.990234 15.986328 A 1.0001 1.0001 0 0 0 32.292969 16.292969 L 25 23.585938 L 17.707031 16.292969 A 1.0001 1.0001 0 0 0 16.990234 15.990234 A 1.0001 1.0001 0 0 0 16.292969 17.707031 L 23.585938 25 L 16.292969 32.292969 A 1.0001 1.0001 0 1 0 17.707031 33.707031 L 25 26.414062 L 32.292969 33.707031 A 1.0001 1.0001 0 1 0 33.707031 32.292969 L 26.414062 25 L 33.707031 17.707031 A 1.0001 1.0001 0 0 0 32.990234 15.986328 z"
                  />
                </svg>
              </div>
            </div>
            <h3
              style="text-align: center;"
            >
              Переключение темы
            </h3>
            <div
              style="display: flex; justify-content: center; align-items: center; font-size: 0.75rem; font-weight: 500;"
            >
              <div>
                cветлая
              </div>
              <div>
                <span
                  class="MuiSwitch-root MuiSwitch-sizeMedium css-lk9ido-MuiSwitch-root"
                >
                  <span
                    class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
                  >
                    <input
                      class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                      type="checkbox"
                    />
                    <span
                      class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                    />
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                  />
                </span>
              </div>
              <div>
                темная
              </div>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      data-testid="header"
    >
      <div
        data-testid="react-logo"
      >
        <img
          alt="Логотип React. На котором пишет Petr Tcoi"
          data-nimg="1"
          decoding="async"
          height="50"
          loading="lazy"
          src="/images/icons/logo-react.svg"
          srcset="/images/icons/logo-react.svg 1x, /images/icons/logo-react.svg 2x"
          style="color: transparent; filter: invert(51%) sepia(85%) saturate(2665%) hue-rotate(322deg) brightness(111%) contrast(99%);"
          title="Логотип React. На котором хорошо пишет Petr Tcoi"
          width="50"
        />
      </div>
      <div
        class="header-title"
        data-testid="header-title"
      >
        Petr Tcoi
      </div>
      <div
        data-testid="popupmenu-burger-button"
      >
        <div
          aria-label="Кнопка открыть меню"
          class="icon-button"
        >
          <svg
            height="25"
            viewBox="0 0 100 75"
            width="25"
          >
            <rect
              height="15"
              width="100"
            />
            <rect
              height="15"
              width="100"
              y="30"
            />
            <rect
              height="15"
              width="100"
              y="60"
            />
          </svg>
        </div>
      </div>
      <div
        class="popup-menu"
        data-popupmenu-status="closed"
        data-testid="popup-menu"
        style="opacity: 0;"
      >
        <div>
          <div
            data-testid="popupmenu-close-button"
            style="display: inline-block;"
          >
            <div
              class="icon-button"
            >
              <svg
                height="35"
                viewBox="0 0 50 50"
                width="35"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M 25 2 C 12.309534 2 2 12.309534 2 25 C 2 37.690466 12.309534 48 25 48 C 37.690466 48 48 37.690466 48 25 C 48 12.309534 37.690466 2 25 2 z M 25 4 C 36.609534 4 46 13.390466 46 25 C 46 36.609534 36.609534 46 25 46 C 13.390466 46 4 36.609534 4 25 C 4 13.390466 13.390466 4 25 4 z M 32.990234 15.986328 A 1.0001 1.0001 0 0 0 32.292969 16.292969 L 25 23.585938 L 17.707031 16.292969 A 1.0001 1.0001 0 0 0 16.990234 15.990234 A 1.0001 1.0001 0 0 0 16.292969 17.707031 L 23.585938 25 L 16.292969 32.292969 A 1.0001 1.0001 0 1 0 17.707031 33.707031 L 25 26.414062 L 32.292969 33.707031 A 1.0001 1.0001 0 1 0 33.707031 32.292969 L 26.414062 25 L 33.707031 17.707031 A 1.0001 1.0001 0 0 0 32.990234 15.986328 z"
                />
              </svg>
            </div>
          </div>
          <h3
            style="text-align: center;"
          >
            Переключение темы
          </h3>
          <div
            style="display: flex; justify-content: center; align-items: center; font-size: 0.75rem; font-weight: 500;"
          >
            <div>
              cветлая
            </div>
            <div>
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium css-lk9ido-MuiSwitch-root"
              >
                <span
                  class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
                >
                  <input
                    class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                    type="checkbox"
                  />
                  <span
                    class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                  />
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                />
              </span>
            </div>
            <div>
              темная
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
